test/hts-precompile/token-managment/tokenManagmentContract.js

```diff
@@ -254,4 +255,693 @@ describe("TokenManagmentContract tests", function () {
+            expect((await txBeforeInfo.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode).to.eq(TX_SUCCESS_CODE);
+            return tokenInfo
+        };
+
+        async function updateTokenInfo(contract, token, updateInfo) {
+            const txUpdate = await contract.updateTokenInfoPublic(token, updateInfo);
+            expect((await txUpdate.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode).to.be.equal(TX_SUCCESS_CODE);
+        }
+
+        function updateTokenInfoValues(keyValueType, key) {
+            const updatedKey = [
+                false,
+                '0x0000000000000000000000000000000000000000',
+                '0x',
+                '0x',
+                '0x0000000000000000000000000000000000000000',
+            ];
+
+            switch (keyValueType) {
+                case utils.KeyValueType.CONTRACT_ID:
+                    updatedKey[1] = key;
+                    break;
+                case utils.KeyValueType.SECP256K1:
+                    updatedKey[3] = key;
+                    break;
+                case utils.KeyValueType.DELEGETABLE_CONTRACT_ID:
+                    updatedKey[4] = key;
+                    break;
+                default:
+                    break;
+            }
+
+            return updatedKey;
+        }
+
+        describe('Admin key set to ECDSA_secp256k', function() {
+            before(async function () {
+                tokenAddress = await utils.createFungibleTokenWithSECP256K1AdminKeyAssociateAndTransferToAddress(tokenCreateContract, tokenCreateContract.address, utils.getSignerCompressedPublicKey());
+                tokenInfoBefore = await getTokenInfo(tokenQueryContract, tokenAddress);
+        
+                await utils.associateToken(tokenCreateContract, tokenAddress, 'TokenCreateContract');
+                await utils.grantTokenKyc(tokenCreateContract, tokenAddress);
+            });
+
+            describe('Positive', function() {
+                it('should be able to change PAUSE key to contractId and pause the token with same contract', async function() {
+                    //Update token info
+                    {
+                        const contractId = tokenManagmentContract.address;
+                        const updatedKey = updateTokenInfoValues(utils.KeyValueType.CONTRACT_ID, contractId);
+    
+                        const token = {
+                            ...tokenInfoBefore, tokenKeys: [[ utils.KeyType.PAUSE, updatedKey]]
+                        };
+                        token.treasury = signers[0].address;
+    
+                        await updateTokenInfo(tokenManagmentContract, tokenAddress, token);
+                    }
+                    
+                    //Pause and unpause token
+                    {
+                        const pauseTokenTx = await tokenManagmentContract.connect(signers[1]).pauseTokenPublic(tokenAddress);
```

Comment:
Is there a way we can know the the token has been paused/unpaused beyond the SUCCESS return value?

Reply:
Yep, I will add those and request review again.

Reply:
Done.